src/Lean/Server/Rpc/Basic.lean

```diff
@@ -42,10 +42,13 @@ non-JSON-serializable fields can be auto-encoded in two ways:
 -- TODO(WN): for Lean.js, have third parameter defining the client-side structure;
 -- or, compile `WithRpcRef` to "opaque reference" on the client
 class RpcEncoding (α : Type) (β : outParam Type) where
-  rpcEncode {m : Type → Type} [Monad m] [MonadRpcSession m] : α → m β
+  rpcEncode {m : Type → Type} [Monad m] [MonadRpcSession m] : α → ExceptT String m β
```

Comment:
This is a bit ugly, but unfortunately necessary to define the unconditional `Inhabited (RpcEncoding α β)` instance below.  The instance is in turn necessary to define `RpcEncoding` instances using `partial def` (which is the easiest approach to support nested inductives).

Reply:
(See comment on the unconditional instance).